q2_gneiss/plot/_plot.py

```diff
@@ -134,8 +134,28 @@ def f(x):
             sample_palette = pd.Series(
                 sns.color_palette("Set2", len(c.value_counts())),
                 index=c.value_counts().index)
-            balance_boxplot(balance_name, data, y=c.name, ax=ax,
-                            palette=sample_palette)
+            try:
+                balance_boxplot(balance_name, data, y=c.name, ax=ax,
+                                palette=sample_palette)
+            except TypeError as e:
+                if 'flexible type' in str(e):
+                    def is_float(category):
+                        try:
+                            float(category)
+                            return True
+                        except ValueError:
+                            return False
+                    categories = data[metadata.name]
+                    inv_categories = list(set(category for category
+                                          in categories if is_float(category)))
+                    inv_categories.sort()
+                    raise ValueError('Your categorical metadata column '
```

Comment:
Can you please confirm, is it a problem if there are _any_ numerical values? I suspect the problem is actually only present if the values are _all_ numerical.

Reply:
You would be correct, that didn't occur to me, but it looks like if any of the values are non-numerical it does work. Makes sense because I suppose if matplotlib gets a value it can definitively assert is not a number it won't worry about whether the rest should be. Though I still don't think that's very useful because it seems unlikely that people would be mixing numerical and nonnumerical categories. I'll add a test for that though.

Reply:
> seems unlikely that people would be mixing numerical and nonnumerical categories

You would be surprised at what people are capable of!

Reply:
Do we only want it to raise an exception if all of their columns are numerical?

Reply:
> Do we only want it to raise an exception if all of their columns are numerical?

Do you mean "values", instead of "columns"? If so, then I think my answer is yes. Rephrased, we should find the minimum set of conditions that produce this error, and check for exactly those conditions, before raising our own error. Make sense?

Reply:
Oh, yes that is what I meant. Yes that does